patches/hardhat@2.23.0.patch

```diff
@@ -1,98 +1,9 @@
---- /dev/null
-+++ b/internal/core/jsonrpc/types/authorization-list.js
-@@ -0,0 +1,38 @@
-+"use strict";
-+var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
-+    if (k2 === undefined) k2 = k;
-+    var desc = Object.getOwnPropertyDescriptor(m, k);
-+    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
-+      desc = { enumerable: true, get: function() { return m[k]; } };
-+    }
-+    Object.defineProperty(o, k2, desc);
-+}) : (function(o, m, k, k2) {
-+    if (k2 === undefined) k2 = k;
-+    o[k2] = m[k];
-+}));
-+var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
-+    Object.defineProperty(o, "default", { enumerable: true, value: v });
-+}) : function(o, v) {
-+    o["default"] = v;
-+});
-+var __importStar = (this && this.__importStar) || function (mod) {
-+    if (mod && mod.__esModule) return mod;
-+    var result = {};
-+    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
-+    __setModuleDefault(result, mod);
-+    return result;
-+};
-+Object.defineProperty(exports, "__esModule", { value: true });
-+exports.rpcAuthorizationList = void 0;
-+const t = __importStar(require("io-ts"));
-+const base_types_1 = require("./base-types");
-+const rpcAuthorizationListTuple = t.type({
-+    chainId: base_types_1.rpcQuantity,
-+    address: base_types_1.rpcAddress,
-+    nonce: base_types_1.rpcQuantity,
-+    yParity: base_types_1.rpcParity,
-+    r: base_types_1.rpcHash,
-+    s: base_types_1.rpcHash,
-+});
-+exports.rpcAuthorizationList = t.array(rpcAuthorizationListTuple);
-+//# sourceMappingURL=authorization-list.js.map
-\ No newline at end of file
-diff --git a/internal/core/jsonrpc/types/authorization-list.js.map b/internal/core/jsonrpc/types/authorization-list.js.map
-new file mode 100644
-index 0000000000000000000000000000000000000000..b612090f21ac898f2701ce3d94e8894adc50af41
---- /dev/null
-+++ b/internal/core/jsonrpc/types/authorization-list.js.map
-@@ -0,0 +1 @@
-+{"version":3,"file":"authorization-list.js","sourceRoot":"","sources":["../../../../src/internal/core/jsonrpc/types/authorization-list.ts"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,yCAA2B;AAE3B,6CAA2E;AAE3E,MAAM,yBAAyB,GAAG,CAAC,CAAC,IAAI,CAAC;IACvC,OAAO,EAAE,wBAAW;IACpB,OAAO,EAAE,uBAAU;IACnB,KAAK,EAAE,wBAAW;IAClB,OAAO,EAAE,sBAAS;IAClB,CAAC,EAAE,oBAAO;IACV,CAAC,EAAE,oBAAO;CACX,CAAC,CAAC;AAEU,QAAA,oBAAoB,GAAG,CAAC,CAAC,KAAK,CAAC,yBAAyB,CAAC,CAAC"}
-\ No newline at end of file
 diff --git a/internal/hardhat-network/provider/provider.d.ts b/internal/hardhat-network/provider/provider.d.ts
-index 087ee3ce522a5791760ed61d0870e87f8a4b01dc..38cd0ea1c9c5dbcb9b5fe4748a3564bee3ab6413 100644
+index 8fd0e3c63ee1e169a8c80ca82ac9c478c951390e..3073c64c187effa96306f07232b55e273f69ca02 100644
```

Comment:
Just for my understanding: what does this patch do?

Reply:
It converts to the multi-chain API that calls `createProvider` on the `EdrContext`